neat/trees/greenstree.py

```diff
@@ -133,12 +120,21 @@ def _setImpedanceDistal(self):
         Set the boundary condition at the distal end of the segment
         """
         if len(self.child_nodes) == 0:
-            self.z_distal = np.infty*np.ones(len(self.z_m)) if self.g_shunt < 1e-10 else \
-                            1. / self.g_shunt
+            z_aux = np.ones(self.z_m.shape, dtype=float)
+
+            if self.g_shunt > 1e-10:
+                z_aux /= self.g_shunt
+            else:
+                z_aux *= np.infty
+
+            self.z_distal = z_aux.astype(self.z_m.dtype)
         else:
-            self.z_distal = 1. / (np.sum([1. / cnode._collapseBranchToRoot() \
-                                         for cnode in self.child_nodes], 0) + \
-                                  self.g_shunt)
+            g_aux = np.ones_like(self.z_m) * self.g_shunt
+
+            for cnode in self.child_nodes:
+                g_aux = g_aux +  1. / cnode._collapseBranchToRoot()
+
+            self.z_distal = 1. / g_aux
```

Comment:
don't we also need to take care of dividing by zero here if `g_shunt` can be zero (~l125)?

Reply:
Great question! The division by zero yields `np.inf`, which is the intended results. I've added a commit `0b2ca8f` that implements uses a context manager to disable the warning